tests/contrib/names/test_names_datastreams.py

```diff
@@ -89,3 +124,23 @@ def test_orcid_http_reader(_, bytes_xml_entry):
     assert len(results) == 1
     assert bytes_xml_entry.entry == results[0].entry
     assert not results[0].errors
+
+
+def test_names_service_writer_create(app, expected_from_xml):
+    writer = NamesServiceWriter("rdm-names", system_identity)
+    record = writer.write(StreamEntry(expected_from_xml))
+    record = record.to_dict()
+
+    assert dict(record, **expected_from_xml) == record
```

Comment:
comment: something weird is happening here, the `IdentifierSet` is stripping out the `scheme` and the url:

```shell
E         Differing items:
E         {'identifiers': [{'identifier': 'https://orcid.org/0000-0001-8135-3489', 'scheme': 'orcid'}]} != {'identifiers': [{'identifier': '0000-0001-8135-3489', 'scheme': 'orcid'}]}
```

Reply:
Fixed, it was due to the `IdentifierScheme` saving a normalized (non-URL) value. Then the subsequent searches and comparissons would be comparing the URL to the normalized value.